app/controllers/Subscriptions.scala

```diff
@@ -128,14 +128,18 @@ class Subscriptions(
 
   def premiumTierGeoRedirect: Action[AnyContent] = geoRedirect("subscribe/premium-tier")
 
-  def displayForm(countryCode: String): Action[AnyContent] =
+  def displayForm(countryCode: String, displayCheckout: String): Action[AnyContent] =
     authenticatedAction(recurringIdentityClientId) { implicit request =>
-      implicit val settings: Settings = settingsProvider.settings()
-      val title = "Support the Guardian | Digital Subscription"
-      val id = "digital-subscription-checkout-page-" + countryCode
-      val js = "digitalSubscriptionCheckoutPage.js"
-      val css = "digitalSubscriptionCheckoutPageStyles.css"
-      Ok(views.html.main(title, id, js, css)).withSettingsSurrogateKey
+      if (displayCheckout == "true") {
+        implicit val settings: Settings = settingsProvider.settings()
+        val title = "Support the Guardian | Digital Subscription"
```

Comment:
We should declare these constants at object level so that they are created once only  throughout the lifecycle of the application. At the moment they are created everytime this function is invoked.